### DISCUSSION TITLE
Licensing clarification for past versions

### DISCUSSION THREAD

**Discussion by asiekierka** [Mystcraft/Mystcraft-Legacy#6] [Q&A]
In light of https://github.com/Mystcraft/Mystcraft-Legacy/commit/3da8544188403967033a13ad4fdd116e48eb428d - does the LGPLv3 license change also apply retroactively to past versions of MystCraft? If not, are their redistribution/modification permissions affected in any way?

Personally, I am curious about 1.4.7 versions in particular, as I am collaborating on a modpack project which aims to preserve the 1.4.7-era modding playthrough experience (with patches, QoL improvements, et cetera); a clarification could potentially benefit the development and distribution ease of this project. I am willing to donate my time to help remove licensing issues and port it to [a modern build system](https://github.com/CrackedPolishedBlackstoneBricksMC/voldeloom), if necessary.

**Comment by XCompWiz** [comment]
The license applies going forward, but the application of it to code retroactively is a bit muddy. Generally, I think it applies to the code in this repo, as the first thing in each branch is the license file, but there's some weird grey area to code not on the branches, since those commits lack a license file at all. Thus, starting new branches from those points is weird.

That said, it's mostly messy, not particularly in question. I should probably stuff the license file at the beginning and regenerate the repo. Better sooner than later, I suppose.

This repo doesn't include 1.4.7 versions, as the code that far back isn't something I can distribute for various reasons. Sorry. As such, I'm not able to help you there.
However, I'm willing to offer my support, at least in spirit, in most cases. Hop by the Discord and we can discuss that further.
